generator/src/main/java/org/corfudb/generator/LongevityApp.java

```diff
@@ -0,0 +1,168 @@
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by rmichoud on 7/25/17.
+ */
+
+/**
+ * This Longevity app will stress test corfu using the Load
+ * generator during a given duration.
+ *
+ * I use a blocking queue in order to limit the number of operation
+ * created at any time. Operations as created and consumed as we go.
+ *
+ */
+@Slf4j
+public class LongevityApp {
+    private long durationMs;
+    private String configurationString;
+    private boolean checkPoint;
+    BlockingQueue<Future> appsFutures;
+    BlockingQueue<Operation> operationQueue;
+    CorfuRuntime rt;
+    State state;
+
+    ExecutorService taskProducer;
+    ScheduledExecutorService checkpointer;
+    ExecutorService workers;
+
+    // How much time we live the application hangs once the duration is finished
+    // and the application is hanged
+    final int applicationTimeoutInMs = 20000;
+
+    final int queueCapacity = 1000;
+
+    long startTime;
+    int numberThreads;
+
+    public LongevityApp(long durationMs, int numberThreads, String configurationString, boolean checkPoint) {
+        this.durationMs = durationMs;
+        this.configurationString = configurationString;
+        this.checkPoint = checkPoint;
+        this.numberThreads = numberThreads;
+
+        operationQueue = new ArrayBlockingQueue<>(queueCapacity);
+        rt = new CorfuRuntime(configurationString).connect();
+        state = new State(50, 100, rt);
+
+        taskProducer = Executors.newSingleThreadExecutor();
+
+        workers = Executors.newFixedThreadPool(numberThreads);
+        checkpointer = Executors.newScheduledThreadPool(1);
+    }
+
+    /**
+     * Let the workers finish naturally (thanks to the timer) and then kill
+     * the producer and the checkpointer.
+     *
+     * At the end of the duration, we give some margin for the workers to finish their task before shutting
+     * down the thread pool.
+     * If the application is hung (which can happen when something goes wrong) we will still terminate.
+     */
+    private void waitForAppToFinish() {
+        workers.shutdown();
+        try {
+            workers.awaitTermination(durationMs + applicationTimeoutInMs, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        taskProducer.shutdownNow();
+
+        checkpointer.shutdownNow();
+    }
+
+    /**
+     * Check if we should still execute tasks or if the duration has elapsed.
+     *
+     * @return If we are still within the duration limit
+     */
+    private boolean withinDurationLimit() {
+        boolean result = System.currentTimeMillis() - startTime < durationMs;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)